docs/api/utility-apis.md

```diff
@@ -55,51 +55,112 @@ from any component inside Backstage, including the ones in `@backstage/core`.
+that returns a new API instance.
+
+For example, this is the default `ApiFactory` for the `ErrorApi`:
 
 ```ts
-import {
-  ApiRegistry,
-  createApp,
-  alertApiRef,
-  errorApiRef,
-  AlertApiForwarder,
-  ErrorApiForwarder,
-  ErrorAlerter,
-  ConfigApi,
-} from '@backstage/core';
-
-const apis = (config: ConfigApi) => {
-  const builder = ApiRegistry.builder();
-
-  // The alert API is a self-contained implementation that shows alerts to the user.
-  const alertApi = builder.add(alertApiRef, new AlertApiForwarder());
-
-  // The error API uses the alert API to send error notifications to the user.
-  builder.add(errorApiRef, new ErrorAlerter(alertApi, new ErrorApiForwarder()));
-  return builder.build();
-};
+createApiFactory({
+  api: errorApiRef,
+  deps: { alertApi: alertApiRef },
+  factory: ({ alertApi }) =>
+    new ErrorAlerter(alertApi, new ErrorApiForwarder()),
+});
+```
 
-const app = createApp({
-  apis,
-  // ... other config
+In this example the `errorApiRef` is our API, which encapsulates the `ErrorApi`
+type. The `alertApiRef` is our single dependency, which we give the name
+`alertApi`, and is then passed on to the factory function, which returns an
+implementation of the `ErrorApi`.
+
+The `createApiFactory` function is a thin wrapper that enables TypeScript type
+inference. You may notice that there are no type annotations in the above
+example, and that is because we're able to infer all types from the `ApiRef`s.
+TypeScript will make sure that the return value of the `factory` function
+matches the type embedded in `api`'s `ApiRef`, in this case the `ErrorApi`. It
+will also match the types between the `deps` and the parameters of the `factory`
+function, again using the type embedded within the `ApiRef`s.
+
+## Registering API Factories
+
+The responsibility for adding Utility APIs to a Backstage app lies in three
+different locations: the Backstage core library, each plugin included in the
+app, and the app itself.
+
+### Core APIs
+
+Starting with the Backstage core library, it provides implementation for all of
+the core APIs. The core APIs are the ones exported by `@backstage/core`, such as
+the `errorApiRef` and `configApiRef`. You can find a full list of them
+[here](../reference/utility-apis/README.md).
+
+The core APIs are loaded for any app created with `createApp` from
+`@backstage/core`, which means that there is no step that needs to be taken to
+include these APIs in an app.
+
+### Plugin APIs
+
+In addition to the core APIs, plugins can define and export their own APIs.
+While doing so they should usually also provide default implementations of their
+own APIs, for example, the `catalog` plugin exports `catalogApiRef`, and also
+supplies a default `ApiFactory` of that API using the `CatalogClient`. There is
+one restriction to plugin-provided API Factories: plugins may not supply
+factories for core APIs, trying to do so will cause the app to crash.
```

Comment:
crash -> refuse to start, or something? Indicating that it's intentional and properly taken care of :)